lib/XTR/MaintenanceRegularTask.php

```diff
@@ -405,34 +405,33 @@ private function publishLayouts()
 
         // check if we have any layouts with set publish date
         if (count($layouts) > 0) {
-
             foreach ($layouts as $layout) {
-
                 // check if the layout should be published now according to the date
                 if (Carbon::createFromTimestamp($layout->publishedDate)->format('U') < Carbon::now()->format('U')) {
                     try {
-                        // check if draft is valid
-                        if ($layout->status === ModuleWidget::$STATUS_INVALID && isset($layout->statusMessage)) {
-                            throw new GeneralException(__($layout->statusMessage));
-                        } else {
-                            // publish the layout
-                            $layout = $this->layoutFactory->concurrentRequestLock($layout, true);
-                            try {
-                                $draft = $this->layoutFactory->getByParentId($layout->layoutId);
-                                $draft->publishDraft();
-                                $draft->load();
-                                $draft->xlfToDisk([
-                                    'notify' => true,
-                                    'exceptionOnError' => true,
-                                    'exceptionOnEmptyRegion' => false
-                                ]);
-                            } finally {
-                                $this->layoutFactory->concurrentRequestRelease($layout, true);
+                        // publish the layout
+                        $layout = $this->layoutFactory->concurrentRequestLock($layout, true);
+                        try {
+                            $draft = $this->layoutFactory->getByParentId($layout->layoutId);
+                            if ($draft->status === ModuleWidget::$STATUS_INVALID
+                                && isset($draft->statusMessage)
+                                && (count($draft->getStatusMessage()) > 1 || count($draft->getStatusMessage()) === 1 && !in_array(__('Empty Region'), $draft->getStatusMessage()))
```

Comment:
I see what you've done here and why, but the "magic" nature of it worries me. For example if we ever change the "Empty Region" error message, this then breaks.

We may need to give this some more thought - an idea which comes to mind (completely unverified) is whether we call `xlfToDisk` on the `$layout`. We could catch an error from there easily enough?

Another potential is a helper method which just checks for empty regions and returns a boolean?